File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.py
class Solution:
    def minimumAverage(self, nums: List[int]) -> float:
        nums.sort()
        n = len(nums)
        res = (nums[0] + nums[n-1]) / 2
        for i in range(1,n//2):
            res = min(res, (nums[i] + nums[n-i-1]) / 2)
        return res

File: 3194-minimum-average-of-smallest-and-largest-elements/3194-minimum-average-of-smallest-and-largest-elements.cpp
class Solution {
public:
    double minimumAverage(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();
        int res = (nums[0] + nums[n-1]);
        for (int i = 1; i < n/2; i++)
            res = min(res, (nums[i] + nums[n-i-1]));
        return (float)res/2;
    }
};